### ISSUE TITLE
ibmcloud authentication support

### ISSUE THREAD

**Issue by allmightyspiff** [softlayer/softlayer-python#1315]
Support for ibmcloud IAM identities.

## slcli config setup
Should allow for users to setup with their ibmcloud user/pass like the `ibmcloud cli` does.
The difference would be this setup would still add an API key for further use (ibmcloud just re-auths with a refresh token I believe).

## transport support
rest/xmlrpc transports should support this format.

## Process

Described a bit here: https://sldn.softlayer.com/article/authenticating-softlayer-api/

basically query  https://iam.cloud.ibm.com/identity/ to get the iam_token
use that token to create an API key
setup that api key in the ~/.softlayer file

**Comment by markododa** [comment]
For a new ibm cloud, is the python script  at the end necessary for creating a softlayer user?
I only managed to login with an api key and "apikey" as username only after succesfully running that script.

**Comment by allmightyspiff** [comment]
The script at the end of https://sldn.softlayer.com/article/authenticating-softlayer-api/? No, It shouldn't be neccessary.

An API key created here ( https://cloud.ibm.com/iam/apikeys ) should work in the latest versions of the SLCLI (with the username "apikey").

An API created here ( https://cloud.ibm.com/iam/users/<USERID>?tab=userdetails) is the legacy style, where you would authenticate with the api username, and api key. Currently it is the only style of authentication the SLCLI supports.
![image](https://user-images.githubusercontent.com/7408017/90403942-4c06f880-e067-11ea-8113-f5f2441c19e8.png)

**Comment by markododa** [comment]
Yes, using apikey with the new api works now, however that started working after i ran the script, but i guess i have missed some step. Thank you!
